RIOT-OS/RIOT
# [tester] IEEE 802.15.4 SubMAC on top of IEEE 802.15.4 Radio HAL

### Contribution description
This draft provides a  __very__ rough implementation of https://github.com/RIOT-OS/RIOT/issues/13376. In a nutshell, this module provides a uniform layer to send and receive IEEE 802.15.4 MAC frames with retransmissions and CSMA-CA. The SubMAC emulates in software hardware features (CSMA-CA, Retransmissions, etc) missing on a given radio.

Besides that, I added a a `netdev_ieee802154_submac` as a transition layer. With this, `default` and `gnrc_networking` are indirectly working on top on the Radio HAL (#13943 ).

I hope this server as a proof for the Radio HAL API :)

I had to include the work by @miri64 from  https://github.com/RIOT-OS/RIOT/pull/11263. Since the radios are asynchronous now, there are some lost fragments when the radio returns `-EBUSY` during the TX process.




### Testing procedure
Check `tests/ieee802154_submac` for the "raw SubMAC API" calls. This example supports sending and receiving data with retransmissions and CSMA-CA.
A more interesting test is running `default` or `gnrc_networking` with `netdev_ieee802154_submac`. With that, all radios from #14655 (CC2538, nrf802154, at86rf2xx) are working with full CSMA-CA/retransmissions support!! :)
Try adding some noise and pinging between nodes.




### Issues/PRs references
#13943 
#13376
#14655 (base)
#11263
